kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/TopicsService.java

```diff
@@ -180,6 +187,25 @@
         .map(clusterMapper::toTopic);
   }
 
+  public Mono<TopicDTO> recreateTopic(KafkaCluster cluster, String topicName) {
+    return adminClientService.get(cluster)
+            .flatMap(cl -> loadTopic(cluster, topicName))
+            .flatMap(topic -> deleteTopic(cluster, topicName).thenReturn(new TopicCreationDTO()
+                            .name(topic.getName())
+                            .partitions(topic.getPartitionCount())
+                            .replicationFactor(topic.getReplicationFactor())
+                            .configs(topic
+                                    .getTopicConfigs()
+                                    .stream()
+                                    .collect(Collectors.toMap(InternalTopicConfig::getName,
+                                            InternalTopicConfig::getValue))))
+                    .delayElement(Duration.ofSeconds(recreateTopicTimeout))
```

Comment:
Is `delayElement` required? Considering we have `retryWhen` lower.

Reply:
maybe this is because it is very unlikely that topic will be immediately deleted. But this timeout is way to high imho

Reply:
Yes, that's right, the topic can't be immediately deleted. Timeout has been decreased to 1 sec.